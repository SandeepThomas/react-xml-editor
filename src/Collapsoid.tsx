import * as React from 'react';

import { IElement } from './Parser';

interface IProps {
    elements: IElement[];
}

export default class Collapsoid extends React.Component<IProps> {
    private static maxLength: number = 35;

    public render(): React.ReactNode {
        const { elements } = this.props;
        return this.renderElements(elements);
    }

    private renderElements(elements: IElement[]): string {
        let collapsoid = '';
        elements.every((element: IElement) => {
            if (element._ && element['#name'] === '__text__') {
                collapsoid += element._ + ' ';
            }
            if (element.$$) {
                collapsoid += this.renderElements(element.$$) + ' ';
            }
            return collapsoid.length < Collapsoid.maxLength;
        });
        return collapsoid.trim().slice(0, Collapsoid.maxLength);
    }
}
