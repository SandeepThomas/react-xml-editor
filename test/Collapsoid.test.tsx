import {describe, expect, jest, test} from '@jest/globals';
import renderer from 'react-test-renderer';

import * as React from 'react';

import Collapsoid from '../src/Collapsoid';

describe('Attribute component', () => {
    test('renders properly', () => {
        const elements = [{
            _: 'text1',
            '#name': '__text__',
        },{
            '$$': [{
                _: 'text2',
                '#name': '__text__',
            }],
            '#name': 'name',
        }];
        const component = renderer.create(
            <Collapsoid
                elements={elements}
            />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
