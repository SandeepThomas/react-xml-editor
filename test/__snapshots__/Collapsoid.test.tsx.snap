// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribute component renders properly 1`] = `"text1 text2"`;
